winrm.go

```diff
@@ -185,7 +196,11 @@ func NewWinRMClient(details getEndpointDetails, options ...winrmSettingsOption)
 	for _, o := range options {
 		client.winrmSettings = o(client.winrmSettings)
 	}
-	client.url = fmt.Sprintf("https://%s:%d/wsman", client.ipAddress, client.port)
+	if client.isSecure {
```

Comment:
It's a detail but to avoid repeating the whole `Sprintf` line, how about:
```
if client.isSecure {
    protocol = "https"
} else {
    protocol = "http"
}
```
then use `protocol`

Reply:
addressed